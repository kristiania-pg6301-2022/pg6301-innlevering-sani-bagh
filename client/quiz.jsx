import {Questions, randomQuestion} from "./questions";
import React from "react";

export function ListQuestions() {
    return <div>
        <h1>All questions with correct answers</h1>
        {Questions.map(q =>
            <div key={q.id}>
                <h1>{q.question}</h1>
                <p>{q.answer}</p>
            </div>
        )}
    </div>;
}

export function Quiz() {
    const [question, setQuestion] = useState(randomQuestion());
    const [answer, setAnswer] = useState();

    if (answer) {
        return <h1>{question.options[answer] === question.answer ? "The answer is correct" : "The answer is wrong"}</h1>;
    }


    return <div>
        <h1>{question.question}</h1>
        {Object.keys(question.options)
            .filter(option => question.options[option])
            .map(option => <p key={option}>
                <button onClick={() => setAnswer(option)}>{question.options[option]}</button>
            </p> )
        }
    </div>;
}