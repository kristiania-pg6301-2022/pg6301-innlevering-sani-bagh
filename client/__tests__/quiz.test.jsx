import {ListQuestions} from "../quiz";
import React from "react";
import ReactDOM  from "react-dom";


describe("list questions", () => {
    it('should show questions list',  () => {
        const element = document.createElement("div");
        ReactDOM.render(<ListQuestions/>, element);
        expect(element.querySelector("h1").innerHTML)
            .toEqual("All questions with correct answers");
        expect(element.innerHTML).toMatchSnapshot();
    });
});