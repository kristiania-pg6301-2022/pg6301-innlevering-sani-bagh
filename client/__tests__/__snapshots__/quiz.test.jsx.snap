// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list questions should show questions list 1`] = `"<div><h1>All questions with correct answers</h1><div><h1>What is the correct JavaScript syntax to change the content of the HTML element below?</h1><p>document.getElementById(\\"demo\\").innerHTML = \\"Hello World!\\";</p></div><div><h1>What is the correct syntax for referring to an external script called \\"xxx.js\\"?</h1><p>&lt;script src=\\"xxx.js\\"&gt;</p></div><div><h1>Which operator is used to assign a value to a variable?</h1><p>=</p></div><div><h1>How to insert a comment that has more than one line?</h1><p>/*This comment has more than one line*/</p></div><div><h1>How do you round the number 7.25, to the nearest integer?</h1><p>Math.round(7.25)</p></div><div><h1>How do you find the number with the highest value of x and y?</h1><p>Math.max(x, y)</p></div><div><h1>How to write an IF statement in JavaScript?</h1><p>if (i == 5)</p></div><div><h1>How does a FOR loop start?</h1><p>for (i = 0; i &lt;= 5; i++)</p></div><div><h1>How does a WHILE loop start?</h1><p>while (i &lt;= 10)</p></div><div><h1>What is the correct way to write a JavaScript array?</h1><p>var colors = [\\"red\\", \\"green\\", \\"blue\\"]</p></div></div>"`;

exports[`list questions should show questions list 1`] = `"<div><h1>All questions with correct answers</h1><div><h1>What is the correct JavaScript syntax to change the content of the HTML element below?</h1><p>document.getElementById(\\"demo\\").innerHTML = \\"Hello World!\\";</p></div><div><h1>What is the correct syntax for referring to an external script called \\"xxx.js\\"?</h1><p>&lt;script src=\\"xxx.js\\"&gt;</p></div><div><h1>Which operator is used to assign a value to a variable?</h1><p>=</p></div><div><h1>How to insert a comment that has more than one line?</h1><p>/*This comment has more than one line*/</p></div><div><h1>How do you round the number 7.25, to the nearest integer?</h1><p>Math.round(7.25)</p></div><div><h1>How do you find the number with the highest value of x and y?</h1><p>Math.max(x, y)</p></div><div><h1>How to write an IF statement in JavaScript?</h1><p>if (i == 5)</p></div><div><h1>How does a FOR loop start?</h1><p>for (i = 0; i &lt;= 5; i++)</p></div><div><h1>How does a WHILE loop start?</h1><p>while (i &lt;= 10)</p></div><div><h1>What is the correct way to write a JavaScript array?</h1><p>var colors = [\\"red\\", \\"green\\", \\"blue\\"]</p></div></div>"`;
